packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt

```diff
@@ -76,7 +76,9 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
         md.update(wideVineId)
         return md.digest().toHexString()
       } catch (e: Exception) {
-        return null
+        // generate random identifier that does not persist across installations
+        // use it as the fallback in case DRM API failed to generate one.
+          return UUID.randomUUID().toString()
```

Comment:
This wasn't exactly what I meant but this also works. Adjust the method signature then since this always returns a String not nullable

Reply:
@oscb it's still possible for this method to return null- 

``` if (collectDeviceId) {
      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2)
        return null
```

I also have the logic here to account for the null possibilities: 

``` if (deviceId !== null) {
      contextInfo.putString("deviceId", deviceId)
    }
```

Thoughts?

Reply:
It's weird that we handle differently wether we can't generate a code because of a failure with MediaDRM or because you are on an old version of Android.

I think it would be better to have the same fallback for both. 

This function should always return null if it can't create it, handle in the caller the scenario where this returns null and `collectDeviceId` is set